From 59e6bcdcea9c8bff99d1c8e754d308acfb596236 Mon Sep 17 00:00:00 2001
From: Julien Thomas <jthomas@zenetys.com>
Date: Mon, 28 Feb 2022 22:13:31 +0100
Subject: [PATCH] BUILD: ssl: another build warning on LIBRESSL_VERSION_NUMBER

We had several warnings when building haproxy 2.5.4 with
old openssl 1.0.1e. This version of openssl is the latest
available in EOL centos 6.

  include/haproxy/openssl-compat.h:157:51: \
  warning: "LIBRESSL_VERSION_NUMBER" is not defined

This patch fixed the build. It changes the #if condition,
as done in other similar parts of openssl-compat.h.
---
 include/haproxy/openssl-compat.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/haproxy/openssl-compat.h b/include/haproxy/openssl-compat.h
index 12a8711..87bb510 100644
--- a/include/haproxy/openssl-compat.h
+++ b/include/haproxy/openssl-compat.h
@@ -174,7 +174,7 @@ static inline int SSL_SESSION_set1_id_context(SSL_SESSION *s, const unsigned cha
 #endif
 
 
-#if (HA_OPENSSL_VERSION_NUMBER < 0x1000200fL) && (LIBRESSL_VERSION_NUMBER < 0x2070500fL)
+#if (HA_OPENSSL_VERSION_NUMBER < 0x1000200fL) && (!defined(LIBRESSL_VERSION_NUMBER) || LIBRESSL_VERSION_NUMBER < 0x2070500fL)
 /* introduced in openssl 1.0.2 */
 
 static inline STACK_OF(X509) *X509_chain_up_ref(STACK_OF(X509) *chain)
-- 
1.7.10.4

